### PR TITLE
torch frontend ViewTensor class Update

### PR DESCRIPTION
**Pull request by JerryGCDing**

### CODE REVIEW

File: ivy/functional/frontends/torch/tensor.py [hunk 1]
```diff
@@ -23,61 +23,61 @@ def add(self, other, *, alpha=1):
...
+        self.data = self.add(other, alpha=alpha).data
         return self.data
 
     def asin(self):
         return torch_frontend.asin(self.data)
 
     def asin_(self):
-        self.data = self.asin()
+        self.data = self.asin().data
         return self.data
```
Comment (djl11) [inline ivy/functional/frontends/torch/tensor.py:34]: Why exactly do these return `self.data`? Should they not return `self`? The current implementation will return an `ivy.Array` which is wrong I believe. Can you change these to all return `self` instead?

File: ivy/functional/frontends/torch/tensor.py [hunk 2]
```diff
@@ -23,61 +23,61 @@ def add(self, other, *, alpha=1):
...
 
     def atan_(self):
-        self.data = self.atan()
+        self.data = self.atan().data
         return self.data
 
     def view(self, shape):
-        self.data = torch_frontend.reshape(self.data, shape)
-        return self.data
+        view = torch_frontend.ViewTensor(weakref.ref(self), shape=shape)
```
Comment (djl11) [inline ivy/functional/frontends/torch/tensor.py:79]: this can be done in a single line

**Review by djl11 — CHANGES_REQUESTED** [review]
haven't dived too deep but looks good to me. Could you please add unit tests which verify that the view tensor behaves as expected? Feel free to implement this test using pytest.parameterize rather than fully generalized via hypothesis. Thanks!

**Comment by JerryGCDing** [comment]
Added a `pytest.mark` keyword check in the fixture within `confest.py` file for frontend tests so that it doesn't prevent parameterized tests from running. The behavior test design for `ViewTensor` class is that for non-inplace instance methods, the test will check if the returned values are the same as its base tensor. For inplace methods, the test will check if the base tensor is being properly updated.

**Review by djl11 — APPROVED** [review]
looks good to me! I'll also just request a review from Mustafa to verify that the changes to `conftest.py` are good, thanks!

**Review by CatB1t — CHANGES_REQUESTED** [review]
Hello @JerryGCDing, I've pushed a fix for `conftest.py` which should run all tests normally, you won't need to use a mark for it, feel free to merge after removing the markers.

**Pull request merged**